test/test_transforms_v2_refactored.py

```diff
@@ -0,0 +1,326 @@
+                dispatcher_param = next(dispatcher_params)
+        except StopIteration:
+            raise AssertionError(
+                f"Parameter `{kernel_param.name}` of kernel `{kernel.__name__}` "
+                f"has no corresponding parameter on the dispatcher `{dispatcher.__name__}`."
+            ) from None
+
+        assert dispatcher_param == kernel_param
+
+
+def _check_dispatcher_datapoint_signature_match(dispatcher):
+    dispatcher_signature = inspect.signature(dispatcher)
+    dispatcher_params = list(dispatcher_signature.parameters.values())[1:]
+
+    datapoint_method = getattr(datapoints._datapoint.Datapoint, dispatcher.__name__)
+    datapoint_signature = inspect.signature(datapoint_method)
+    datapoint_params = list(datapoint_signature.parameters.values())[1:]
+
+    # Because we use `from __future__ import annotations` inside the module where `datapoints._datapoint` is
+    # defined, the annotations are stored as strings. This makes them concrete again, so they can be compared to the
+    # natively concrete dispatcher annotations.
+    datapoint_annotations = get_type_hints(datapoint_method)
+    for param in datapoint_params:
+        param._annotation = datapoint_annotations[param.name]
+
+    assert dispatcher_params == datapoint_params
+
+
+def check_dispatcher_signatures_match(dispatcher, *, kernel, datapoint_type):
+    _check_dispatcher_kernel_signature_match(dispatcher, kernel=kernel, datapoint_type=datapoint_type)
+    _check_dispatcher_datapoint_signature_match(dispatcher)
+
+
+def check_transform():
+    pass
+
+
+class TestResize:
+    @pytest.mark.parametrize("size", [(11, 17), (15, 13)])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_image_tensor(self, size, antialias, dtype, device):
+        image = make_image(size=(14, 16), dtype=dtype, device=device)
+        check_kernel(F.resize_image_tensor, image, size=size, antialias=antialias)
+
+    @pytest.mark.parametrize("size", [(11, 17), (15, 13)])
+    @pytest.mark.parametrize("format", list(datapoints.BoundingBoxFormat))
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.int64])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_kernel_bounding_box(self, size, format, dtype, device):
+        spatial_size = (14, 16)
+        bounding_box = make_bounding_box(format=format, spatial_size=spatial_size, dtype=dtype, device=device)
+        check_kernel(F.resize_bounding_box, bounding_box, spatial_size=spatial_size, size=size)
+
+    @pytest.mark.parametrize("kernel", [F.resize_image_tensor, F.resize_bounding_box])
+    @pytest.mark.parametrize("size", [(11, 17), (15, 13)])
+    @pytest.mark.parametrize("antialias", [True, False])
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.uint8, torch.int64])
+    @pytest.mark.parametrize("device", cpu_and_gpu())
```

Comment:
Is it absolutely needed to call `check_dispatcher` over all those inputs?
Sounds like a lot of the checks aren't relaly input-dependent and yet will call the kernel over all those inputs like e.g. the `torch._C._log_api_usage_once` check. It sounds wasteful?

Reply:
We had errors in the early days of v2 for two reasons:
1. `torch.jit.script(dispatcher)` went fine, but failed at runtime.
2. We some minimal parameter wrangling inside the dispatcher and that was buggy, but went undetected, because we didn't actually called the kernel

For these reasons we opted to just do the full matrix again. But I fully agree that we are wasting resources by checking the logging multiple times.

Reply:
> when were those issue detected? In other tests, or e.g. during e.g. execution of the training scripts?

I honestly can't tell you anymore without digging. I know that they went undetected and we either stumbled over them during training or doing something related.

> Basically: how much risk is there that we publish a release with one of those issues before catching it?

I would say the risk is low maybe pushing medium here. We removed most of data wrangling from the dispatchers and kernels are fully tested for CUDA and JIT.

Let's reduce the matrix here. If we actually stumble over this again, we can always increase.

Reply:
See 0f124862cdabd068c05080d98bddfeacf5538951